exports/dataset.cpp

```diff
@@ -492,6 +492,18 @@ using as_VariableView =
                         std::string, boost::container::small_vector<double, 8>,
                         Dataset, Eigen::Vector3d>;
 
+class SubsetHelper {
```

Comment:
Maybe SubsetFacade or SubsetWrapper would be better as a name

Reply:
Aren't "facade" as well as "Wrapper" misleading? It is not a facade for a subset, and not a wrapper for a subset. It is a helper for creating subsets. Anyway, it is just used locally, so unless this is used more widely...

Reply:
Ok, i'm not particularly concerned about this. It was this https://github.com/mantidproject/dataset/pull/77/files#diff-ed677a2708c9afda3b34f34bde8de2b0R814 that prompted me to raise the question.